### PR TITLE
MaterialDatePicker now implements HasValue<Date>

### PR DESCRIPTION
**Pull request by gilberto-torrezan**
MaterialDatePicker now implements HasValue<Date>, and contains handlers for when the popup opens and closes. Removed the experimental delegate interface to handle the events.

### CODE REVIEW

File: gwt-material/src/main/java/gwt/material/design/client/ui/MaterialDatePicker.java [hunk 1]
```diff
@@ -293,10 +274,10 @@ public native void setPickerDate(JsDate date, Element picker) /*-{
     }-*/;
 
     public Date getDate() {
-        return getPickerDate();
+        return getDate();
```
Comment (anton-johansson) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialDatePicker.java]: Stack overflow? :)

Comment (gilberto-torrezan) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialDatePicker.java]: Well spotted!

**Comment by kevzlou7979** [comment]
Thanks for the PR. I will test this out and will create a showcase about the implementation.

**Comment by kevzlou7979** [comment]
Showcase created :)

**Pull request merged**